hazelcast/src/test/java/com/hazelcast/instance/HazelcastInstanceManagerTest.java

```diff
@@ -43,62 +39,80 @@
 import static org.mockito.Mockito.doThrow;
 
 @RunWith(HazelcastParallelClassRunner.class)
-@Category({QuickTest.class, ParallelTest.class})
+@Category(QuickTest.class)
 public class HazelcastInstanceManagerTest extends HazelcastTestSupport {
 
-    @Test
-    public void testTestHazelcastInstanceFactory() {
-        TestHazelcastInstanceFactory instanceFactory = new TestHazelcastInstanceFactory();
-        final HazelcastInstance instance1 = instanceFactory.newHazelcastInstance();
-        final HazelcastInstance instance2 = instanceFactory.newHazelcastInstance();
-        final HazelcastInstance instance3 = instanceFactory.newHazelcastInstance();
-
+    private HazelcastInstance hazelcastInstance;
 
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() throws Exception {
-                assertEquals(3, instance1.getCluster().getMembers().size());
-                assertEquals(3, instance2.getCluster().getMembers().size());
-                assertEquals(3, instance3.getCluster().getMembers().size());
-            }
-        });
+    @After
+    public void tearDown() {
+        if (hazelcastInstance != null) {
+            hazelcastInstance.shutdown();
+        }
+    }
 
-        instanceFactory.terminateAll();
+    @AfterClass
+    public static void cleanUp() {
+        HazelcastInstanceManager.terminateAll();
+    }
 
+    @Test
+    public void testConstructor() {
+        assertUtilityConstructor(HazelcastInstanceManager.class);
     }
 
     @Test
-    public void testTestHazelcastInstanceFactory_withTwoFactories() {
+    public void testTestHazelcastInstanceFactory() {
         TestHazelcastInstanceFactory instanceFactory = new TestHazelcastInstanceFactory();
-        final HazelcastInstance instance1 = instanceFactory.newHazelcastInstance();
-        final HazelcastInstance instance2 = instanceFactory.newHazelcastInstance();
-        final HazelcastInstance instance3 = instanceFactory.newHazelcastInstance();
+        try {
+            final HazelcastInstance instance1 = instanceFactory.newHazelcastInstance();
+            final HazelcastInstance instance2 = instanceFactory.newHazelcastInstance();
+            final HazelcastInstance instance3 = instanceFactory.newHazelcastInstance();
+
+            assertTrueEventually(new AssertTask() {
+                @Override
+                public void run() throws Exception {
+                    assertEquals(3, instance1.getCluster().getMembers().size());
+                    assertEquals(3, instance2.getCluster().getMembers().size());
+                    assertEquals(3, instance3.getCluster().getMembers().size());
+                }
+            });
+        } finally {
+            instanceFactory.terminateAll();
```

Comment:
I moved the `TestHazelcastInstanceFactory` shutdown to a finally block to ensure it's called even an exception is thrown in the test. Since it's not used in all tests, I didn't pull it out as class field.